packages/datagrid/src/datagrid.ts

```diff
@@ -0,0 +1,3701 @@
+   */
+  get baseRowSize(): number {
+    return this._rowSections.baseSize;
+  }
+
+  /**
+   * Set the base size of the body rows.
+   *
+   * #### Notes
+   * This is the size of rows which have not been resized.
+   */
+  set baseRowSize(value: number) {
+    this._setBaseSize(this._rowSections, value);
+  }
+
+  /**
+   * Get the base size of the body columns.
+   *
+   * #### Notes
+   * This is the size of columns which have not been resized.
+   */
+  get baseColumnSize(): number {
+    return this._columnSections.baseSize;
+  }
+
+  /**
+   * Set the base size of the body columns.
+   *
+   * #### Notes
+   * This is the size of columns which have not been resized.
+   */
+  set baseColumnSize(value: number) {
+    this._setBaseSize(this._columnSections, value);
+  }
+
+  /**
+   * Get the base size of the row headers.
+   *
+   * #### Notes
+   * This is the size of row headers which have not been resized.
+   */
+  get baseRowHeaderSize(): number {
+    return this._rowHeaderSections.baseSize;
+  }
+
+  /**
+   * Set the base size of the row headers.
+   *
+   * #### Notes
+   * This is the size of row headers which have not been resized.
+   */
+  set baseRowHeaderSize(value: number) {
+    this._setBaseSize(this._rowHeaderSections, value);
+  }
+
+  /**
+   * Get the base size of the column headers.
+   *
+   * #### Notes
+   * This is the size of column headers which have not been resized.
+   */
+  get baseColumnHeaderSize(): number {
+    return this._columnHeaderSections.baseSize;
+  }
+
+  /**
+   * Set the base size of the column headers.
+   *
+   * #### Notes
+   * This is the size of column headers which have not been resized.
+   */
+  set baseColumnHeaderSize(value: number) {
+    this._setBaseSize(this._columnHeaderSections, value);
+  }
+
+  /**
+   * Get the size of a section in the data grid.
+   *
+   * @param area - The grid area for the section of interest.
+   *
+   * @param index - The index of the section of interest.
+   *
+   * @return The size of the section, or `-1` if `index` is invalid.
+   */
+  sectionSize(area: 'row' | 'column' | 'row-header' | 'column-header', index: number): number {
+    return this._getSectionList(area).sectionSize(index);
+  }
+
+  /**
+   * Resize a section in the data grid.
+   *
+   * @param area - The grid area for the section of interest.
+   *
+   * @param index - The index of the section of interest.
+   *
+   * @param size - The new size for the section.
+   *
+   * #### Notes
+   * This is a no-op if `index` is invalid.
+   */
+  resizeSection(area: 'row' | 'column' | 'row-header' | 'column-header', index: number, size: number): void {
+    this._resizeSection(this._getSectionList(area), index, size);
+  }
+
+  /**
+   * Reset sections in the data grid to their base size.
+   *
+   * @param area - The grid area for the section of interest.
+   */
+  resetSections(area: 'row' | 'column' | 'row-header' | 'column-header'): void {
```

Comment:
No. Because it's not taking an index. It's resetting *all* sections withing that area.